QEfficient/utils/_utils.py

```diff
@@ -219,7 +219,16 @@ def get_padding_shape_from_config(config, batch_size, seq_len):
     ):  # Check for num_key_value_heads (Llama/Mistral)
         n_heads = config.num_key_value_heads
         d_head = config.hidden_size // config.num_attention_heads
-    elif hasattr(config, "n_heads"):  # Check for n_heads and d_model in the config (MPT Model)
+    elif (
```

Comment:
isn't this condition same as line 231-233?
We can move those line here, and remove 223-226?

Reply:
Actually no, the MPT model needed more specific parameters. Because simply testing for `n_heads` causes dbrx also satisfy the condition. MPT and dbrx has similar `config.json`. Hence current check fails, and wrong config will be set for the model.